### PR TITLE
feat(apps/interface): support theming on rainbowkit

### PR DESCRIPTION
**Pull request by asaadam**
It's a bit verbose. 

but I cannot find a simpler solution for this.

### CODE REVIEW

File: apps/interface/pages/_app.tsx [hunk 1]
```diff
@@ -100,7 +108,10 @@ function App({ Component, pageProps }: AppProps) {
             />
             <ChakraProvider theme={themes}>
                 <WagmiConfig client={wagmiClient}>
-                    <RainbowKitProvider chains={chains}>
+                    <RainbowKitProvider
+                        chains={chains}
+                        theme={theme === "dark" ? darkTheme() : undefined}
```
Comment (pyk) [inline apps/interface/pages/_app.tsx:113]: @asaadam why we don't use `useColorMode` directly here? see example here https://github.com/risedle/monorepo/blob/540e6d0235b4594646416495ada46365d78cec2b/apps/interface/components/DarkmodeToggle/index.tsx#L7-L15

**Review by pyk — CHANGES_REQUESTED** [review]

File: apps/interface/pages/_app.tsx (hunk 1 above)
Comment (asaadam) [reply apps/interface/pages/_app.tsx:113]: because it's outside of chakra provider ser 

CC: @math-marcellino

Comment (asaadam) [reply apps/interface/pages/_app.tsx:113]: if you're okay with this implementation, i will continue to create test for this PR

Comment (pyk) [reply apps/interface/pages/_app.tsx:113]: @asaadam please check my comment on linear issue. I think it's better to use container component so `useColorMode` can work

Comment (asaadam) [reply apps/interface/pages/_app.tsx:113]: oh ya that solution also can be work. 
i will close this MR

**Pull request closed without merging**